Kudu.Core/Jobs/JobsFileWatcher.cs

```diff
@@ -0,0 +1,231 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using Kudu.Contracts.Jobs;
+using Kudu.Contracts.Settings;
+using Kudu.Core.Infrastructure;
+using Kudu.Core.Tracing;
+
+namespace Kudu.Core.Jobs
+{
+    public class JobsFileWatcher : IDisposable
```

Comment:
It would be really great if for each of these classes you'd add code comments detailing what the class responsibilities are.